src/ddprof_worker.cc

```diff
@@ -168,15 +168,24 @@ DDRes ddprof_pr_sample(DDProfContext *ctx, perf_event_sample *sample, int pos) {
   unsigned long this_ticks_unwind = __rdtsc();
   DDRes res = unwindstate__unwind(us);
 
+  // Usually we want to send the sample_val, but sometimes we need to process
+  // the event to get the desired value
+  PerfWatcher *watcher = &ctx->watchers[pos];
+  uint64_t sample_val = sample->period;
+  if (PERF_SAMPLE_RAW & watcher->sample_type) {
+    uint64_t raw_offset = watcher->trace_off;
```

Comment:
This is pretty neat in terms of design. The watcher knows where to find the value and keeps an offset. 
I guess in some cases, it could make sense to have fixed values, though we could adapt that.
I wonder if the sample->period could be "merged" some way with this design. I did not give it any thought.